Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -793,7 +793,9 @@ public class PBXProjGenerator {
             }
 
             // Set INFOPLIST_FILE
-            if let info = target.info {
+            if project.targetHasBuildSetting("INFOPLIST_FILE", target: target, config: config) {
+                // do not overwrite Info.plist path when already defined
+            } else if let info = target.info {
```

Comment:
This could actually go inside the `if !project.targetHasBuildSetting("INFOPLIST_FILE", target: target, config: config)` below